src/containerapp/azext_containerapp/commands.py

```diff
@@ -0,0 +1,109 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+from azure.cli.core.commands import CliCommandType
+from msrestazure.tools import is_valid_resource_id, parse_resource_id
+from azext_containerapp._client_factory import ex_handler_factory
+
+
+def transform_containerapp_output(app):
+    props = ['name', 'location', 'resourceGroup', 'provisioningState']
+    result = {k: app[k] for k in app if k in props}
+
+    try:
+        result['fqdn'] = app['properties']['configuration']['ingress']['fqdn']
+    except Exception:
+        result['fqdn'] = None
+
+    return result
+
+
+def transform_containerapp_list_output(apps):
+    return [transform_containerapp_output(a) for a in apps]
+
+
+def transform_revision_output(rev):
+    props = ['name', 'active', 'createdTime', 'trafficWeight']
+    result = {k: rev['properties'][k] for k in rev['properties'] if k in props}
+
+    if 'name' in rev:
+        result['name'] = rev['name']
+
+    if 'fqdn' in rev['properties']['template']:
+        result['fqdn'] = rev['properties']['template']['fqdn']
+
+    return result
+
+
+def transform_revision_list_output(revs):
+    return [transform_revision_output(r) for r in revs]
+
+
+def load_command_table(self, _):
+    with self.command_group('containerapp') as g:
+        g.custom_command('show', 'show_containerapp', table_transformer=transform_containerapp_output)
+        g.custom_command('list', 'list_containerapp', table_transformer=transform_containerapp_list_output)
+        g.custom_command('create', 'create_containerapp', supports_no_wait=True, exception_handler=ex_handler_factory())
+        g.custom_command('update', 'update_containerapp', supports_no_wait=True, exception_handler=ex_handler_factory())
+        g.custom_command('delete', 'delete_containerapp', exception_handler=ex_handler_factory())
```

Comment:
Do we need to consider adding `confirmation=true` for those delete commands?

Reply:
Added as per your suggestion